### PR TITLE
Tool 14 rewrite project helper

### PR DESCRIPTION
**Pull request by BirmacherAkos**
Current project_helper.rb:
https://github.com/bitrise-steplib/steps-ios-auto-provision/blob/master/lib/autoprovision/project_helper.rb

Missing parts:
- force_code_sign_properties(): It will be big change, so I want to do it in a separate PR.

--

**Changes**
- autoprovision/projecthelper.go
- autoprovision/projecthelper_test.go
- dependency management: dep

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -4,5 +4,9 @@
   unused-packages = true
 
 [[constraint]]
-  branch = "master"
+  branch = "project-attribute"
```
Comment (BirmacherAkos) [inline Gopkg.toml]: Temporarily. Do not forget to change it back to master.

File: autoprovision/projecthelper_test.go [hunk 2]
```diff
@@ -0,0 +1,665 @@
...
+	configCases = []string{
+		"Debug",
+		"Release",
+		"Debug",
+		"Release",
+		"Debug",
+		"Release",
+	}
+	projectCases = []string{
+		"/Users/akosbirmacher/Develop/go/src/github.com/bitrise-steplib/steps-ios-auto-provision/_tmp/Xcode-10_default.xcworkspace",
```
Comment (BirmacherAkos) [inline autoprovision/projecthelper_test.go]: Do not forget to upload them to the sample repo, and update the path.

Comment (BirmacherAkos) [reply autoprovision/projecthelper_test.go]: Fixed

File: autoprovision/projecthelper.go [hunk 3]
```diff
@@ -1 +1,452 @@
...
+	if _, err := plist.Unmarshal(b, &options); err != nil {
+		return "", fmt.Errorf("failed to unmarshal Info.plist, error: %s ", err)
+	}
+
+	bundleID, ok := options["CFBundleIdentifier"].(string)
+	if !ok || bundleID == "" {
+		return "", fmt.Errorf("failed to parse CFBundleIdentifier from the Info.plist")
+	}
+
+	log.Warnf("CFBundleIdentifier defined with variable: %s, trying to resolve it...", bundleID)
```
Comment (lszucs) [inline autoprovision/projecthelper.go]: Did you forget to check whether the value is a variable reference at this point?

Original implementation checks and returns at this point, if possible: https://github.com/bitrise-steplib/steps-ios-auto-provision/blob/master/lib/autoprovision/project_helper.rb#L169

File: autoprovision/projecthelper.go [hunk 4]
```diff
@@ -1 +1,452 @@
...
+
+	captures := r.FindString(bundleID)
+
+	prefix := strings.Split(captures, "$")[0]
+	envKey := strings.Split(strings.SplitAfter(captures, "(")[1], ":")[0]
+	suffix := strings.Join(strings.SplitAfter(captures, ")")[1:], "")
+
+	envValue, err := buildSettings.String(envKey)
+	if err != nil {
+		return "", fmt.Errorf("failed to find enviriment variable value for key %s, error: %s", envKey, err)
```
Comment (lszucs) [inline autoprovision/projecthelper.go]: typo: environment

File: autoprovision/projecthelper.go (hunk 3 above)
Comment (BirmacherAkos) [reply autoprovision/projecthelper.go]: fixed

File: autoprovision/projecthelper.go (hunk 4 above)
Comment (BirmacherAkos) [reply autoprovision/projecthelper.go]: fixed

**Comment by BirmacherAkos** [comment]
Temporarily disabled the test:
https://github.com/bitrise-steplib/steps-ios-auto-provision/pull/66/files#diff-618c7096679001171c100a6ff5d12efaR85

Need to talk about it with @godrei

**Pull request merged**